src/components/PrivateRoute.jsx

```diff
@@ -1,12 +1,27 @@
 import React from 'react';
 import {Route, Redirect} from 'react-router-dom';
 import {useSelector} from 'react-redux'
+import {paths} from '../utilities/constants';
 
 import {selectAuthIsLoggedIn} from '../redux/selectors/authSelector';
+import {hasAccessToPath} from '../utilities/auth';
 
-export default function PrivateRoute({ component: Component, ...rest }) {
+export default function PrivateRoute({ component: Component, path, ...rest }) {
   const isLoggedIn = useSelector(selectAuthIsLoggedIn);
-  const routeComponent = isLoggedIn ? Component : () => <Redirect to='/auth/login'/>;
 
-  return <Route {...rest} component={routeComponent} />
+  function getComponent() {
+    if (!isLoggedIn) {
+      return () => <Redirect to={paths.LOGIN} />;
+    }
+
+    if (hasAccessToPath(path)) {
+      return Component;
+    }
+
+    return () => <Redirect to={paths.PROFILE} />
```

Comment:
Could you explain when and why we want to redirect user to `profile` view?

Reply:
This is the only view that all the Roles can access.